templates/user/settings/keys_principal.tmpl

```diff
@@ -1,12 +1,8 @@
-{{if .AllowPrincipals}}
+{{if and .AllowPrincipals (not DisableSSH)}}
 	<h4 class="ui top attached header">
 		{{.i18n.Tr "settings.manage_ssh_principals"}}
 		<div class="ui right">
-		{{if not .DisableSSH}}
 			<div class="ui blue tiny show-panel button" data-panel="#add-ssh-principal-panel">{{.i18n.Tr "settings.add_new_principal"}}</div>
-		{{else}}
-			<div class="ui blue tiny button disabled">{{.i18n.Tr "settings.ssh_disabled"}}</div>
```

Comment:
I think for the user's information these should be kept.

Reply:
There should be no unnecessary items in UI present to make it clean as possible. If owner disables SSH in the application there is no point in leaving SSH items in UI.

Reply:
SSH is de-facto standard for Git operations nowadays and is expected to be supported universally everywhere. User should most definitely be informed that SSH is not available.

Reply:
> I would open a bug report that this menu is not present,

Reading system manual should be enough.

There should be no unnecessary items in UI present to make it as clean as possible. We prefer to give users clean UI without unnecessary information noise and will hide all SSH stuff. Feel free to close this PR if you don't like it.

> and is expected to be supported universally everywhere

We don't expect using git over SSH so the above is false :).

Reply:
>We don't expect using git over SSH so the above is false :).

Your singular needs do not represent the bigger picture. Please understand that Gitea is a project aimed at far bigger audience - for a feature to be added it should take into consideration that fact. I see no bigger issues with this PR other than the mentioned notification about disabled SSH being removed.

Reply:
> Your singular needs[...]

We only push PRs that may be valuable for bigger audience. Clean IU is IHMO. As mentioned above - fell free to close this PR if you don't like it.